src/Components/v2/CarouselV2.tsx

```diff
@@ -0,0 +1,272 @@
+import { Box, ChevronIcon, Flex, space } from "@artsy/palette"
+import { Options as FlickityOptions } from "flickity"
+import React, { ReactNode, useRef, useState } from "react"
+import styled from "styled-components"
+import { left, LeftProps, right, RightProps } from "styled-system"
+import { useDidMount } from "Utils/hooks/useDidMount"
+import { Media } from "Utils/Responsive"
+
+type Arrow = (
+  props: {
+    Arrow: React.ReactType
+    getFlickity: any
+  } & Props
+) => React.ReactNode
+
+interface Props {
+  height?: number
+  data: object[] // This is designed to handle any shape of data passed, as long as its an array
+  render: (slide) => ReactNode
+  renderLeftArrow?: Arrow
+  renderRightArrow?: Arrow
+  onArrowClick?: () => void
+  options?: FlickityOptions
+}
+
+const isClient = typeof window !== "undefined"
+
+/*
+ * Flickity docs https://flickity.metafizzy.co
+ * Flickity requires window, so we must render a seperate Flex wrapper instead of
+ * a carousel if we're using SSR.
+ */
+export class Carousel extends React.Component<Props> {
+  static defaultProps = {
+    height: 300,
+  }
+
+  render() {
+    return (
+      <>
+        {isClient && (
+          <Box key={Math.random()}>
+            <Media at="xs">
+              <SmallCarousel {...this.props} />
+            </Media>
+            <Media greaterThan="xs">
+              <LargeCarousel {...this.props} />
+            </Media>
+          </Box>
+        )}
+        {!isClient && (
+          <Box key={Math.random()}>
```

Comment:
I don't think you should leave the `Math.random` here. These keys just need to be different from each other so the `client` and `server` strings _should_ work. It's just that with `Math.random` there's a _slight_ chance that they may end up with the same values with would cause a really hard to reproduce bug.

Reply:
the `client` and `server` strings weren't working but i think we could do `client + math.random()` or an equivalent so they'll never match.

Reply:
Is a problem to use consistent keys? A `isServer ? 'keyA' : 'keyB'` sort of thing?

Otherwise, i don't know. Theoretically its possible for `Math.random()` to return two identical numbers two times in a row, but in that case the person seeing the missing slider probably wasn't meant to see the slider... That said, it's kinda a 🤔 to have to go the `Math.random()` route -- might want to play with it a bit.

Reply:
Seems weird to me that the two different strings aren't working. Random with a prefix would be fine. 👍